packages/compose/src/hooks/use-viewport-match/index.js

```diff
@@ -53,8 +65,15 @@ const CONDITIONS = {
  * @return {boolean} Whether viewport matches query.
  */
 const useViewportMatch = ( breakpoint, operator = '>=' ) => {
-	const mediaQuery = `(${ CONDITIONS[ operator ] }: ${ BREAKPOINTS[ breakpoint ] }px)`;
-	return useMediaQuery( mediaQuery );
+	const simulatedWidth = useContext( ViewportMatchWidthContext );
+	const mediaQuery = ! simulatedWidth && `(${ CONDITIONS[ operator ] }: ${ BREAKPOINTS[ breakpoint ] }px)`;
+	const mediaQueryResult = useMediaQuery( mediaQuery );
+	if ( simulatedWidth ) {
+		return OPERATOR_EVALUATORS[ operator ]( BREAKPOINTS[ breakpoint ], simulatedWidth );
+	}
+	return mediaQueryResult;
 };
 
+useViewportMatch.__experimentalWidthProvider = ViewportMatchWidthContext.Provider;
```

Comment:
Is there any prior art for how we assign context as properties of hooks? Wondering if it should be its own exported member instead.

Reply:
No, I think we never did this before. I exported as a property of the hook to make clear that the component only affects the hook return value. But I can export an __experimentalUseViewportMatchWidthProvider  component instead. I don't have a preference here. If you have a preference for the component version, let me know, and I will perform the update.

Reply:
Hm. The main thing that caught my eye was in how it's used:

```jsx
<useViewportMatch.__experimentalWidthProvider value={ undefined } />
```

Technically it's valid, but a little non-conventional.

For me, if it was a separate thing, I don't think we would want to associate it so strongly to being only relevant for this hook, so I would imagine something like `ViewportWidthProvider` being sufficient.

But especially given that this package is meant to be used exclusively for these sorts of helpers (higher-order components and hooks), I think it would be fair to associate it directly with the hook.

For that reason, I think what you've proposed here can be okay.